src/main/java/com/cloudbees/jenkins/plugins/awscredentials/AmazonWebServicesCredentialsBinding.java

```diff
@@ -57,24 +57,28 @@
 
     public final static String DEFAULT_ACCESS_KEY_ID_VARIABLE_NAME = "AWS_ACCESS_KEY_ID";
     private final static String DEFAULT_SECRET_ACCESS_KEY_VARIABLE_NAME = "AWS_SECRET_ACCESS_KEY";
-    private final static String SESSION_TOKEN_VARIABLE_NAME = "AWS_SESSION_TOKEN";
+    private final static String DEFAULT_SESSION_TOKEN_VARIABLE_NAME = "AWS_SESSION_TOKEN";
 
     @NonNull
     private final String accessKeyVariable;
     @NonNull
     private final String secretKeyVariable;
+    @NonNull
+    private final String sessionTokenVariable;
 
     /**
      *
      * @param accessKeyVariable if {@code null}, {@value DEFAULT_ACCESS_KEY_ID_VARIABLE_NAME} will be used.
      * @param secretKeyVariable if {@code null}, {@value DEFAULT_SECRET_ACCESS_KEY_VARIABLE_NAME} will be used.
+     * @param sessionTokenVariable if {@code null}, {@value DEFAULT_SESSION_TOKEN_VARIABLE_NAME} will be used.
      * @param credentialsId
      */
     @DataBoundConstructor
-    public AmazonWebServicesCredentialsBinding(@Nullable String accessKeyVariable, @Nullable String secretKeyVariable, String credentialsId) {
+    public AmazonWebServicesCredentialsBinding(@Nullable String accessKeyVariable, @Nullable String secretKeyVariable, @Nullable String sessionTokenVariable, String credentialsId) {
```

Comment:
We have to keep the existing constructors to maintain binary compatibility. Instead of changing constructor, add new ones (moving the `@DataBoundConstructor` to the appropriate one) or use a `@DataBoundSetter` for the new field.